### ISSUE TITLE
[Bug]: [GoSDK] After loading partial fields successfully, searching and output "*" fields reports an error

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#37853]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20241118-12ed40e1-amd64
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): go sdk
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. collection has many fields
2. load partial fields: 
```
	partialLoadedFields := []string{common.DefaultInt64FieldName, common.DefaultVarcharFieldName, common.DefaultFloatVecFieldName, common.DefaultFloat16VecFieldName}
```
3. search and output "*", sdk parser output fields failed:
```
"extra output fields [int32Array int8Array int16Array int64Array floatArray doubleArray int8 int16 binaryVec varcharArray bf16Vec bool int32 boolArray json float double] found and result does not dynamic field"]
```

- case:
```
func TestLoadPartialFields(t *testing.T) {
	/*
		1.  verify the collection loaded successfully
		2.  verify the loaded fields can be searched in expr and output_fields
		3.  verify the skip fields not loaded, and cannot search/query/hybrid search with them in expr or output_fields
	*/
	// create collection -> insert -> flush -> index
	ctx := hp.CreateContext(t, time.Second*common.DefaultTimeout)
	mc := createDefaultMilvusClient(ctx, t)

	prepare, schema := hp.CollPrepare.CreateCollection(ctx, t, mc, hp.NewCreateCollectionParams(hp.AllFields), hp.TNewFieldsOption(), hp.TNewSchemaOption())
	prepare.InsertData(ctx, t, mc, hp.NewInsertParams(schema), hp.TNewDataOption())
	prepare.FlushData(ctx, t, mc, schema.CollectionName)
	prepare.CreateIndex(ctx, t, mc, hp.TNewIndexParams(schema))

	// load partial fields
	partialLoadedFields := []string{common.DefaultInt64FieldName, common.DefaultVarcharFieldName, common.DefaultFloatVecFieldName, common.DefaultFloat16VecFieldName}
	loadTask, err := mc.LoadCollection(ctx, clientv2.NewLoadCollectionOption(schema.CollectionName).WithLoadFields(partialLoadedFields...).WithSkipLoadDynamicField(true))
	common.CheckErr(t, err, true)
	err = loadTask.Await(ctx)
	common.CheckErr(t, err, true)

	// search loaded fields & loaded fileds expr & output loaded fields
	expr := fmt.Sprintf("%s > '2' ", common.DefaultVarcharFieldName)
	vectors := hp.GenSearchVectors(common.DefaultNq, common.DefaultDim, entity.FieldTypeFloatVector)

	searchRes, err := mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithOutputFields("*").WithFilter(expr))
	common.CheckErr(t, err, true)
	common.CheckSearchResult(t, searchRes, common.DefaultNq, common.DefaultLimit)
	log.Debug("TestLoadPartialFields", zap.Any("searchRes", searchRes[0].Err))
	common.CheckOutputFields(t, partialLoadedFields, searchRes[0].Fields)

	// search with not-loaded anns field -> Error
	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultBFloat16VecFieldName))
	common.CheckErr(t, err, false, "field not loaded")

	// search with expr not loaded field -> Error
	invalidExpr := fmt.Sprintf("%s > 2.0 ", common.DefaultFloatFieldName)
	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithFilter(invalidExpr))
	common.CheckErr(t, err, false, "field not loaded")

	// search with output_fields not loaded field -> Error
	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithOutputFields(common.DefaultBoolFieldName))
	common.CheckErr(t, err, false, "field not loaded")
}
```

**Comment by torrischen** [comment]
> ### Is there an existing issue for this?
> * [x]  I have searched the existing issues
> 
> ### Environment
> ```
> - Milvus version: master-20241118-12ed40e1-amd64
> - Deployment mode(standalone or cluster):
> - MQ type(rocksmq, pulsar or kafka):    
> - SDK version(e.g. pymilvus v2.0.0rc2): go sdk
> - OS(Ubuntu or CentOS): 
> - CPU/Memory: 
> - GPU: 
> - Others:
> ```
> 
> ### Current Behavior
> 1. collection has many fields
> 2. load partial fields:
> 
> ```
> 	partialLoadedFields := []string{common.DefaultInt64FieldName, common.DefaultVarcharFieldName, common.DefaultFloatVecFieldName, common.DefaultFloat16VecFieldName}
> ```
> 
>  3. search and output "*", sdk parser output fields failed:
> 
> ```
> "extra output fields [int32Array int8Array int16Array int64Array floatArray doubleArray int8 int16 binaryVec varcharArray bf16Vec bool int32 boolArray json float double] found and result does not dynamic field"]
> ```
> 
>  * case:
> 
> ```
> func TestLoadPartialFields(t *testing.T) {
> 	/*
> 		1.  verify the collection loaded successfully
> 		2.  verify the loaded fields can be searched in expr and output_fields
> 		3.  verify the skip fields not loaded, and cannot search/query/hybrid search with them in expr or output_fields
> 	*/
> 	// create collection -> insert -> flush -> index
> 	ctx := hp.CreateContext(t, time.Second*common.DefaultTimeout)
> 	mc := createDefaultMilvusClient(ctx, t)
> 
> 	prepare, schema := hp.CollPrepare.CreateCollection(ctx, t, mc, hp.NewCreateCollectionParams(hp.AllFields), hp.TNewFieldsOption(), hp.TNewSchemaOption())
> 	prepare.InsertData(ctx, t, mc, hp.NewInsertParams(schema), hp.TNewDataOption())
> 	prepare.FlushData(ctx, t, mc, schema.CollectionName)
> 	prepare.CreateIndex(ctx, t, mc, hp.TNewIndexParams(schema))
> 
> 	// load partial fields
> 	partialLoadedFields := []string{common.DefaultInt64FieldName, common.DefaultVarcharFieldName, common.DefaultFloatVecFieldName, common.DefaultFloat16VecFieldName}
> 	loadTask, err := mc.LoadCollection(ctx, clientv2.NewLoadCollectionOption(schema.CollectionName).WithLoadFields(partialLoadedFields...).WithSkipLoadDynamicField(true))
> 	common.CheckErr(t, err, true)
> 	err = loadTask.Await(ctx)
> 	common.CheckErr(t, err, true)
> 
> 	// search loaded fields & loaded fileds expr & output loaded fields
> 	expr := fmt.Sprintf("%s > '2' ", common.DefaultVarcharFieldName)
> 	vectors := hp.GenSearchVectors(common.DefaultNq, common.DefaultDim, entity.FieldTypeFloatVector)
> 
> 	searchRes, err := mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithOutputFields("*").WithFilter(expr))
> 	common.CheckErr(t, err, true)
> 	common.CheckSearchResult(t, searchRes, common.DefaultNq, common.DefaultLimit)
> 	log.Debug("TestLoadPartialFields", zap.Any("searchRes", searchRes[0].Err))
> 	common.CheckOutputFields(t, partialLoadedFields, searchRes[0].Fields)
> 
> 	// search with not-loaded anns field -> Error
> 	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultBFloat16VecFieldName))
> 	common.CheckErr(t, err, false, "field not loaded")
> 
> 	// search with expr not loaded field -> Error
> 	invalidExpr := fmt.Sprintf("%s > 2.0 ", common.DefaultFloatFieldName)
> 	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithFilter(invalidExpr))
> 	common.CheckErr(t, err, false, "field not loaded")
> 
> 	// search with output_fields not loaded field -> Error
> 	_, err = mc.Search(ctx, clientv2.NewSearchOption(schema.CollectionName, common.DefaultLimit, vectors).WithANNSField(common.DefaultFloatVecFieldName).WithOutputFields(common.DefaultBoolFieldName))
> 	common.CheckErr(t, err, false, "field not loaded")
> }
> ```
> 
> ### Expected Behavior
> _No response_
> 
> ### Steps To Reproduce
> _No response_
> 
> ### Milvus Log
> _No response_
> 
> ### Anything else?
> _No response_

Has this been solved? I met this error either.

**Comment by torrischen** [comment]
Has this been solved? I met this problem either.

**Comment by ThreadDao** [comment]
fixed master-20241202-3a7a8c79-amd64
